lib/elixir/lib/kernel/utils.ex

```diff
@@ -22,25 +22,40 @@ defmodule Kernel.Utils do
         _ -> raise ArgumentError, "invalid syntax in defdelegate #{Macro.to_string(fun)}"
       end
 
-    :ok = check_defdelegate_args(args)
-
-    as_args =
-      case append_first and args != [] do
-        true  -> tl(args) ++ [hd(args)]
-        false -> args
+    as_args_list = normalize_args(args)
+    as = Keyword.get(opts, :as, name)
+    Enum.map as_args_list, fn as_args ->
+      formal_args = make_formal_args(as_args)
+      as_args = case append_first do
+        true  -> tl(as_args) ++ [hd(as_args)]
+        false -> as_args
       end
+      {name, formal_args, as, as_args}
+    end
+  end
 
-    as = Keyword.get(opts, :as, name)
-    {name, args, as, as_args}
+  defp make_formal_args(args) do
+    Enum.filter args, fn
+      {name, _, mod} when is_atom(name) and is_atom(mod) -> true
```

Comment:
Thanks!! I wasn't aware that `match?` could be used with guards.
This is clearly way better!